spacy/util.py

```diff
@@ -236,33 +238,46 @@ def get_package_version(name):
         return None
 
 
-def split_version(version):
-    """RETURNS (tuple): Two integers, the major and minor spaCy version."""
-    pieces = version.split(".", 3)
-    return int(pieces[0]), int(pieces[1])
+def is_compatible_version(version, constraint, prereleases=True):
+    """Check if a version (e.g. "2.0.0") is compatible given a version
+    constraint (e.g. ">=1.9.0,<2.2.1"). If the constraint is a specific version,
+    it's interpreted as =={version}.
 
+    version (str): The version to check.
+    constraint (str): The constraint string.
+    prereleases (bool): Whether to allow prereleases. If set to False,
+        prerelease versions will be considered incompatible.
+    RETURNS (bool / None): Whether the version is compatible, or None if the
+        version or constraint are invalid.
+    """
+    # Handle cases where exact version is provided as constraint
+    if constraint[0].isdigit():
+        constraint = f"=={constraint}"
+    try:
+        spec = SpecifierSet(constraint)
+        version = Version(version)
+    except (InvalidSpecifier, InvalidVersion):
+        return None
```

Comment:
What sort of version constraints might be invalid? Trying to figure out whether `None` is better than raising.

Reply:
Something weirdly formatted, something that happens to not be a valid version. It's not about the version constraint, it's about whether the string is considered a valid version.

Reply:
Yeah but should we raise there? I guess if you download a model with a bad version string it's okay to let you load it.

Reply:
I think raising here may cause too many problem – if a version is formatted wrong, we can still load the model and we will report an incompatibility warning. I think it's better to be cautios here.

(Also, I currently have some logic in `spacy validate` that sets the version to `"n/a"` and then expects a `None` compatibility result. But of course I can adjust that if we decide we do want to raise for non-versions.)